File: backend/models/message.ts
import { randomUUID } from "node:crypto";
import { pgTable, text, timestamp, varchar } from "drizzle-orm/pg-core";
import { threads } from "./thread.js";

export const messages = pgTable("messages", {
  id: varchar("id")
    .primaryKey()
    .$defaultFn(() => randomUUID()),
  thread_id: varchar("thread_id").references(() => threads.id),
  role: varchar("role").$type<"user" | "assistant" | "tool">(),
  content: text("content").default(""),
  tool_call_id: varchar("tool_call_id"),
  tool_name: varchar("tool_name"),
  created_at: timestamp("created_at").defaultNow(),
});

File: backend/models/message.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from backend.models import Base


class Message(Base):
    __tablename__ = "messages"

    id: Mapped[str] = mapped_column(
        String, primary_key=True, default=lambda: str(uuid.uuid4())
    )
    thread_id: Mapped[str] = mapped_column(String, ForeignKey("threads.id"))
    role: Mapped[str] = mapped_column(String)  # user, assistant, tool
    content: Mapped[str] = mapped_column(Text, default="")
    tool_call_id: Mapped[str | None] = mapped_column(String, nullable=True)
    tool_name: Mapped[str | None] = mapped_column(String, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    thread = relationship("Thread")
